sympy/solvers/ode/riccati.py

```diff
@@ -0,0 +1,791 @@
+
+    Substitute and multiply by `x^m`.
+
+    >>> g = f.subs(x, x + x0).cancel()
+    >>> num, den = (g*x**m).cancel().as_numer_denom()
+    >>> num
+    1 - x
+    >>> den
+    x - 4
+
+    Form the equation to find the recurrence relation
+
+    ..math :: 1 - x = (x - 4) \Sum_{x = 0}^{\infty} a[i] x^i
+
+    Multiplying this out and rearranging the indices of the
+    sums, we get -
+
+    ..math :: 1 - x = -4 a_0 + (a_0 - 4 a_1) x + \Sum_{x = 2}^{\infty} (a[i-1] - 4*a[i]) x^i
+
+    Find the base case coefficients and then the recurrence
+    relation. Solving for these, we get -
+
+    ..math:: a_0 = -\frac{1}{4}, a_1 = {3}{16}, a_m = \frac{a_{m-1}}{4}
+    """
+    one = Poly(1, x, extension=True)
+
+    if r == oo:
+        # Series at x = oo is equal to first transforming
+        # the function from x -> 1/x and finding the
+        # series at x = 0
+        num, den = inverse_transform_poly(num, den, x)
+        r = S(0)
+
+    if r:
+        # For an expansion about a non-zero point, a
+        # transformation from x -> x + r must be made
+        num = num.transform(Poly(x + r, x, extension=True), one)
+        den = den.transform(Poly(x + r, x, extension=True), one)
+        num, den = num.cancel(den, include=True)
+
+    # Remove the pole from the denominator if the series
+    # expansion is about one of the poles
+    num, den = (num*x**m).cancel(den, include=True)
+
+    # Equate coefficients for the first terms (base case)
+    maxdegree = 1 + max(num.degree(), den.degree())
+    syms = symbols(f'a:{maxdegree}', cls=Dummy)
+    diff = num - den * Poly(syms[::-1], x)
+    coeff_diffs = diff.all_coeffs()[::-1][:maxdegree]
+    (coeffs, ) = linsolve(coeff_diffs, syms)
+
+    # Use the recursion relation for the rest
+    recursion = den.all_coeffs()[::-1]
+    div, rec_rhs = recursion[0], recursion[1:]
+    series = list(coeffs)
+    while len(series) < n:
+        next_coeff = sum(c*series[-1-n] for n, c in enumerate(rec_rhs)) / div
+        series.append(-next_coeff)
+    series = {m - i: val for i, val in enumerate(series)}
+    return series
+
+def compute_m_ybar(x, poles, choice, c, d, N):
+    ybar = 0
+    m = Poly(d[choice[0]][-1], x, extension=True)
+
+    # Calculate the first (nested) summation for ybar
+    # as given in Step 9 of the Thesis (Pg 82)
+    for i in range(len(poles)):
+        for j in range(len(c[i][choice[i + 1]])):
+            ybar += c[i][choice[i + 1]][j]/(x - poles[i])**(j+1)
+        m -= Poly(c[i][choice[i + 1]][0], x, extension=True)
+
+    # Calculate the second summation for ybar
+    for i in range(N+1):
+        ybar += d[choice[0]][i]*x**i
+    return (m.expr, ybar)
+
+
+def solve_aux_eq(numa, dena, numy, deny, x, m):
+    # Assume that the solution is of the type
+    # p(x) = C0 + C1*x + ... + Cm*x**m
+    psyms = symbols(f'C0:{m}', cls=Dummy)
+    domain = ZZ[psyms]
+    psyms += (1, )
+    psol = 0
+    for i in range(m + 1):
+        psol += Poly(psyms[i]*x**i, x, domain=domain)
```

Comment:
A `Poly` can be constructed directly from a list of coefficients:
```python
In [9]: syms = symbols('x:3')

In [10]: Poly(syms, z, domain=ZZ[syms])
Out[10]: Poly(x0*z**2 + x1*z + x2, z, domain='ZZ[x0,x1,x2]')

In [11]: Poly(syms[::-1], z, domain=ZZ[syms])
Out[11]: Poly(x2*z**2 + x1*z + x0, z, domain='ZZ[x0,x1,x2]')
```

Reply:
I want to construct a polynomial of the form `C_0 + C_1*x + C_2*x**2 + ... + C_{m-1}*x**(m - 1) + x**m`, but the below code gives a generator error. It works if I don't specify the domain, but is there a way to specify the domain and make it work, since it takes time to construct the domain of `Poly`?
```
psyms = symbols(f'C0:{m}', cls=Dummy) + ('1', )
psol = Poly(psyms[::-1], x, ZZ[psyms])
```

Reply:
Why are you appending the string `1` to the symbols tuple?

Also domain needs to be given as a keyword argument.

Reply:
Mentioning the kwarg removed the error. The error was saying something about a generator having only string, symbol, etc, so I made it a string. Infact, 1 is not even required to be mentioned since it would already be part of `ZZ`.